exporters/readers/s3_reader.py

```diff
@@ -242,7 +242,6 @@ def get_read_streams(self):
             file_obj = urlopen(key.generate_url(S3_URL_EXPIRES_IN))
             yield Stream(file_obj, key_name, key.size)
 
-    @retry_long
     def read_lines_from_keys(self):
         for current_key in self.keys:
             self.current_key = current_key
```

Comment:
hm, perhaps extracting out a function for the part in the middle that handles only one key, and use retry_long for that bit would be a decent next step.
Don't know about handling `self.last_block` and similar stuff though.